### PR TITLE
Add README for the repo

### PR DESCRIPTION
**Pull request by dzou**
Initial commit of the README for the repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,62 @@
...
+    <repository>
+        <id>snapshots-repo</id>
+        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        <releases><enabled>false</enabled></releases>
+        <snapshots><enabled>true</enabled></snapshots>
+    </repository>
+</repositories>
+```
+
+After adding the snapshots repository, you can add the `google-cloud-graalvm-support` dependency
```
Comment (meltsufin) [inline README.md:23]: Should we follow Spring's 1-line-per-sentence convention?

File: README.md [hunk 2]
```diff
@@ -4,3 +4,62 @@
 
 This repository provides support for applications using the [Google Java Client Libraries](https://github.com/googleapis/google-cloud-java)
 to be built as [GraalVM Native Images](https://www.graalvm.org/docs/reference-manual/native-image).
```
Comment (meltsufin) [inline README.md:6]: The link seems to have changed to: https://www.graalvm.org/reference-manual/native-image/.

File: README.md [hunk 3]
```diff
@@ -4,3 +4,62 @@
...
+    <artifactId>google-cloud-graalvm-support</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+This dependency contains the GraalVM configurations (reflection config, native call config) to
+provide out-of-the-box support for native-image compilation of applications depending on the
+Google Java Client Libraries.
+
+### Note
```
Comment (meltsufin) [inline README.md:38]:
```suggestion
### Note on grpc-netty-shaded version
```

File: README.md [hunk 4]
```diff
@@ -4,3 +4,62 @@
...
+```
+
+The Java client libraries import this dependency to your project automatically as a
+transitive dependency. The latest version of `grpc-netty-shaded` contains a patch to enable native
+image compilation, so this version must be (temporarily) manually upgraded for native image
+compilation to work.
+
+## Samples
+
+Sample projects can be found in the [google-cloud-graalvm-samples](https://github.com/GoogleCloudPlatform/google-cloud-graalvm-support/tree/master/google-cloud-graalvm-samples)
```
Comment (meltsufin) [inline README.md:58]: This sample doesn't seem to have an associated README. Also, does it make sense to add some more instructions here on how to compile and run native-image, or to link to a page that does?

**Review by meltsufin — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dzou) [reply README.md]: Ok, that sounds good. Done.

File: README.md (hunk 2 above)
Comment (dzou) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (dzou) [reply README.md]: Done. Added Readme for sample.

**Review by meltsufin — APPROVED** [review]
LGTM!

**Review by elefeint — APPROVED** [review]
Great docs.

**Pull request merged**